Source/engine.cpp

```diff
@@ -480,22 +480,24 @@ void CelBlitOutlineTo(CelOutputBuffer out, BYTE col, int sx, int sy, BYTE *pCelB
 				if (dst < out.end() && dst > out.begin()) {
 					if (dst >= out.end() - out.pitch()) {
 						while (width) {
-							if (*src++) {
+							if (!skipColorIndexZero || *src > 0) {

```

Comment:
You could change less lines using: `skipColorIndexZero && *src++`

Reply:
I actually prefer the current form in this PR as it doesn't do multiple types of operation in a single line.

Reply:
Someone correct me if I'm wrong, but wouldn't that lead to invalid behaviour? The src pointer needs to increase by one every pixel, but if the "if" statement is `skipColorIndexZero && *src++` then I think what's after "&&" is never ran whenever skipColorIndexZero is false.

Reply:
You could make the suggestion work if you flip it i.e. `*src++ && skipColorIndexZero` But I think the way you have it now makes it the most legible, as post-increment in this case could be confusing.

Reply:
@FluffyQuack you are right, I missed that.